src/Dependencies/captureDependencies.lua

```diff
@@ -52,7 +53,11 @@ local function captureDependencies(saveToSet: Types.Set<Types.Dependency<any>>,
 	-- now that the shared state has been set up, call the callback in a pcall.
 	-- using a pcall means the shared state can be reset afterwards, even if an
 	-- error occurs.
-	local ok, value = xpcall(callback, parseError, ...)
+	local ok, value = xpcall(function() 
+		noYield(function(...) 
+			noYield(callback, ...)
+		end, parseError, ...)
+	end, parseError)
```

Comment:
I feel like this should be simplified - we likely don't need anonymous functions here.

Would passing the arguments inline work? Something like this:
```suggestion
	local ok, value = xpcall(noYield, parseError, callback, ...)
```